#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.41577104278669125 -0.8027209348577601 -0.4275202225873665 1.7081249655321142
  position -0.818920334674369 1.3596463105386776 0.1679168982314309
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.5 1.5
  floorAppearance PBRAppearance {
    roughness 1
    roughnessMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
}
E-puck {
  translation 0.531086 0 0.0653554
  rotation 0 1 0 2.87979
  controller "line_follow"
  groundSensorsSlot [
    DistanceSensor {
      translation 0 0.0175 0
      name "ir7"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.01 0
      name "ir6"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.005 0
      name "ir5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.00125 0
      name "ir4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.00125 0
      name "ir3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.005 0
      name "ir2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0175 0
      name "ir0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.01 0
      name "ir1"
      type "infra-red"
    }
  ]
}
TrackOne {
  translation 0 -0.00753 0
}
Robot {
  translation 0.542661 0.0300009 -0.0162034
  rotation -8.772619999896721e-08 4.851619999942883e-06 0.9999999999882272 3.14159
  children [
    DEF ir_panel Solid {
      translation 0 0.02 -0.028
      rotation 0 0 1 1.5708
      children [
        DistanceSensor {
          translation 0 0.0175 0
          name "ir7"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.01 0
          name "ir6"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.005 0
          name "ir5"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.00125 0
          name "ir4"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.00125 0
          name "ir3"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.005 0
          name "ir2"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.0175 0
          name "ir0"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.01 0
          name "ir1"
          type "infra-red"
        }
        DEF panel Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.003 0.04 0.01
          }
        }
      ]
      name "ir_panel"
      boundingObject USE panel
      physics Physics {
      }
    }
    DEF jack Solid {
      translation 0 0 0.02
      rotation 0 1 0 0.261799
      scale 0.1 0.1 0.1
      children [
        DEF cap Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Capsule {
            height 0.4
            radius 0.05
          }
        }
      ]
      name "solid(2)"
      boundingObject USE cap
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.011 0 -0.019
      rotation 0 1 0 1.8325996938995748
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Capsule {
            height 0.05
            radius 0.005
            subdivision 26
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.011 0 -0.019
      rotation 0 1 0 1.3326
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Capsule {
            height 0.05
            radius 0.005
            subdivision 26
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
      }
      geometry Sphere {
        radius 0.022
        subdivision 4
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.025 0 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
      ]
      endPoint Solid {
        translation 0.025 0 0
        rotation 0 0 1 1.5708
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.005
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics DEF physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.025 0 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint Solid {
        translation -0.025 0 0
        rotation 0 0 1 1.5708
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE physics_wh
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "line_follow"
}
